Documentation for use in on-boarding

dragan opened the pull request:
The initial documentation for Scipian, this by far from being complete, but it gives us a start and also gets things out of my head. 😄 

- [ ] Intro: The What, Why, Who, and How
- [ ] Primitives
- [ ] Architecture

It's never too early to get feedback, creating a draft PR first.

dragan committed:
docs: Introduction

  - Initial documention around the What, Why, Who, and How

Signed-off-by: Dale Ragan <dale.ragan@sap.com>

dragan committed:
docs: Primitives

  - Define the primitives we hope to solve with the happy path
  - These will drive the RFC's

Signed-off-by: Dale Ragan <dale.ragan@sap.com>

ncronquist commented on docs/intro.md:
```diff
@@ -0,0 +1,120 @@
+
+The value stream behind Scipian is to focus on speed and agility - bringing new
+innovation and technology to support business strategy. Scipian is propelling
+the engineer's to build applications by enabling engineers to focus on what
```
sp: engineer's -> engineers

dragan replied:
Feedback addressed in latest rebase.

dragan committed:
docs: Architecture

  - Describes the components of the Scipian architecture and how these
    components integrate with existing architectures, such as
    Kubernetes.

Signed-off-by: Dale Ragan <dale.ragan@sap.com>

dragan committed:
docs: Index page for easy navigation

Signed-off-by: Dale Ragan <dale.ragan@sap.com>

ncronquist commented on docs/intro.md:
```diff
@@ -0,0 +1,120 @@
+like Amazon Web Services, Google Cloud Platform, and Microsfot Azure. Deployment
+is very differnt between the options, which caused the engineering teams to
+create deployment tools, which meant they had to build them and maintain the
+toolchains. This can reach up to 70% of their time to get a deployment out for
```
question: Not arguing with this 70% figure, but is there any data we could link to for it?

dragan replied:
Currently it's an arbitrary number, I removed it for now and added a TODO to replace once researched more with links to back it up. That was the plan anyway, might as well track it that way. 😄

ncronquist commented on docs/intro.md:
```diff
@@ -0,0 +1,120 @@
+is very differnt between the options, which caused the engineering teams to
+create deployment tools, which meant they had to build them and maintain the
+toolchains. This can reach up to 70% of their time to get a deployment out for
+the first time. Then on top of this, there's no standards on how a team can
```
nit: `there's` -> `there are`

dragan replied:
Feedback addressed in latest rebase.

ncronquist commented on docs/intro.md:
```diff
@@ -0,0 +1,120 @@
+create deployment tools, which meant they had to build them and maintain the
+toolchains. This can reach up to 70% of their time to get a deployment out for
+the first time. Then on top of this, there's no standards on how a team can
+implement logging, monitoring, and tracing which is basic operational tasks.
```
`is` -> `are`

dragan replied:
Feedback addressed in latest rebase.

ncronquist reviewed (commented):
Looks like a great start!

christav commented on docs/intro.md:
```diff
@@ -0,0 +1,122 @@
+
+From an product engineering standpoint, teams have a very disparate experience.
+They need to deploy applications to internal data centers and a cloud provider
+like Amazon Web Services, Google Cloud Platform, and Microsfot Azure. Deployment
```
```suggestion
like Amazon Web Services, Google Cloud Platform, and Microsoft Azure. Deployment
```

dragan replied:
Feedback addressed in latest rebase.

christav commented on docs/intro.md:
```diff
@@ -0,0 +1,122 @@
+From an product engineering standpoint, teams have a very disparate experience.
+They need to deploy applications to internal data centers and a cloud provider
+like Amazon Web Services, Google Cloud Platform, and Microsfot Azure. Deployment
+is very differnt between the options, which caused the engineering teams to
```
```suggestion
is very different between the options, which caused the engineering teams to
```

dragan replied:
Feedback addressed in latest rebase.

nicklathe commented on docs/intro.md:
```diff
@@ -0,0 +1,122 @@
+
+The value stream behind Scipian is to focus on speed and agility - bringing new
+innovation and technology to support business strategy. Scipian is propelling
+the engineers to build applications by enabling engineers to focus on what
```
```suggestion
the engineers to build applications by enabling them to focus on what
```

dragan replied:
Feedback addressed in latest rebase.

nicklathe commented on docs/primitives.md:
```diff
@@ -0,0 +1,225 @@
+- Application code is deployed outside of the normal
+straight-from-source-control process.
+- The system becomes finicky. It only accepts deploys in a certain manual way.
+- The init scripts no longer work, unless you so something special and
```
```suggestion
- The init scripts no longer work, unless you do something special and
```

dragan replied:
Feedback addressed in latest rebase.
